### PR TITLE
MOBT-288: Modifications to OccurrenceWithinVicinity to enable the use of multiple radii

### PR DESCRIPTION
**Pull request by bayliffe**
This change modifies all places in which the OccurrenceWithinVicinity plugin is used to provide the vicinity distance as a list. The plugin has been modified to loop through the provided values generating a forecast for each vicinity size. These are combined together into a single cube. Unit tests have been extended to show the handling of multiple vicinity radii.

The threshold and vicinity CLIs have been modified to accept lists of vicinity radii instead of single values. There are additional acceptance tests.

Acceptance test data branch: `mobt288`

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver_tests/utilities/test_OccurrenceWithinVicinity.py [hunk 1]
```diff
@@ -385,12 +385,41 @@ def test_with_multiple_realizations_and_times(
...
+    result = OccurrenceWithinVicinity(radii=[RADIUS], land_mask_cube=land)(cube)
     assert isinstance(result, Cube)
     assert result.data.shape == orig_shape
     assert np.allclose(result.data, expected)
 
 
+@pytest.mark.parametrize(
+    "kwargs", (({"radii": [2000, 5000]}), ({"grid_point_radii": [2, 3, 4, 5]}),)
+)
+def test_coordinate_order_with_multiple_realizations_and_times(
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver_tests/utilities/test_OccurrenceWithinVicinity.py:397]: Does the [test_metadata ](https://github.com/metoppv/improver/blob/665cd32b178235c1257595072843da3b415f1e99/improver_tests/utilities/test_OccurrenceWithinVicinity.py#L155)test not do what you expect?

File: improver_tests/utilities/test_OccurrenceWithinVicinity.py [hunk 2]
```diff
@@ -288,9 +288,9 @@ def test_with_land_mask_and_mask(cube, land_mask_cube):
     mask = np.zeros((5, 5))
     mask[0, 4] = 1
     cube.data = np.ma.array(cube.data, mask=mask)
-    result = OccurrenceWithinVicinity(RADIUS, land_mask_cube=land_mask_cube).process(
-        cube
-    )
+    result = OccurrenceWithinVicinity(
+        radii=[RADIUS], land_mask_cube=land_mask_cube
+    ).process(cube)
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver_tests/utilities/test_OccurrenceWithinVicinity.py:293]: Done.

File: improver_tests/utilities/test_OccurrenceWithinVicinity.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver_tests/utilities/test_OccurrenceWithinVicinity.py:397]: I thought I'd seen it somewhere, then I couldn't find it again.

**Review by MoseleyS — DISMISSED** [review]

File: improver/cli/vicinity.py [hunk 3]
```diff
@@ -59,13 +59,15 @@ def process(cube: cli.inputcube, vicinity: float):
         cube (iris.cube.Cube):
             A cube containing data to which a vicinity it to be applied.
         vicinity (float):
-            Distance in metres used to define the vicinity within which to
-            search for an occurrence
+            List of distances in metres used to define the vicinities within
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/cli/vicinity.py:62]: Fixed, and the type changed from (float) to (list of float / int) as well.

**Review by Kat-90 — APPROVED** [review]
Reviewed and re-ran acceptance tests for change. Happy to approve.

**Pull request merged**